src/util.ts

```diff
@@ -0,0 +1,140 @@
+import { redis } from "./index.js";
+import { Message, Server } from "./types.js";
+
+/**
+ * Synchronizes the servers with the provided list of servers.
+ *
+ * @param servers - The list of servers to synchronize.
+ * @returns A promise that resolves to an object containing the synchronized servers.
+ * @throws If there is an error while synchronizing the servers.
+ */
+export async function syncServers(
+  servers: Server[],
+): Promise<{ servers: Server[] }> {
+  const serversSetKey = "servers";
+  const response: Server[] = [];
+
+  try {
+    const myServerStrings = await redis.sMembers(serversSetKey);
+    const myServers: Server[] = myServerStrings.map((str) => JSON.parse(str));
+
+    for (const server of myServers) {
+      if (!servers.some((s) => s.ip === server.ip && s.port === server.port)) {
+        response.push(server);
+      }
+    }
+
+    for (const server of servers) {
+      if (
+        !myServers.some((s) => s.ip === server.ip && s.port === server.port)
+      ) {
+        await redis.sAdd(serversSetKey, JSON.stringify(server));
+      }
+    }
+
+    return { servers: response };
+  } catch (error) {
+    console.error("Error syncing servers:", error);
+    throw error;
+  }
+}
+
+/**
+ * Synchronizes the hashes with the provided list of hashes.
+ * @param hashes - The list of hashes to synchronize.
+ * @returns A promise that resolves to an object containing the synchronized hashes.
+ * @throws If there is an error while synchronizing the hashes.
+ */
+export async function syncHashes(
+  hashes: string[],
+): Promise<{ hashes: string[] }> {
+  const response: string[] = [];
+
+  try {
+    const myHashes = await redis.keys("hash:*");
+
+    for (const hash of myHashes) {
+      const shortHash = hash.substring(5);
+      if (!hashes.includes(shortHash)) {
+        response.push(shortHash);
+      }
+    }
+
+    return { hashes: response };
+  } catch (error) {
+    console.error("Error syncing hashes:", error);
+    throw error;
+  }
+}
+
+/**
+ * Store a message in Redis.
+ * @param message - The message to store.
+ */
+export function storeMessage(message) {
+  const recipient = message.recipient;
+  const hashKey = `hash:${message.hash}`;
+  const messageStr = JSON.stringify(message);
+  const recipientSetKey = `recipient:${recipient}`;
+  const expiryTime = 7 * 24 * 60 * 60;
+  redis.set(hashKey, messageStr);
+  redis.sAdd(recipientSetKey, hashKey);
+  redis.expire(hashKey, expiryTime);
+  // TODO: remove items from the recipient set as they're expired. This expiry logic does not work and the set will infinitely grow.
```

Comment:
Pls address todo or make a github issue.

Reply:
also didn't write this comment. this update is focused on typescript